v-next/hardhat/test/internal/builtin-plugins/solidity/build-system/cache.ts

```diff
@@ -69,42 +70,44 @@ describe("ObjectCache", () => {
       assert.deepEqual(filesAfter, filesBefore);
     });
 
-    it("should remove everything with the max age set to 0", async () => {
+    it("should remove everything with the max age set to -1", async () => {
       const filesBefore = await getAllFilesMatching(cachePath);
       assert.notDeepEqual(filesBefore, []);
-      // NOTE: We're waiting a little so that the file's atime is different
-      await new Promise((resolve) => setTimeout(resolve, 10));
-      await cache.clean(0);
+      // NOTE: -1 is not enough on Windows for some reason. Maybe there is some
+      // delay in how the access time is set on files on creation?
+      await cache.clean(-100);
       const filesAfter = await getAllFilesMatching(cachePath);
       assert.deepEqual(filesAfter, []);
     });
 
-    it("should remove everything with the max size set to 0", async () => {
+    it("should remove everything with the max size set to -1", async () => {
       const filesBefore = await getAllFilesMatching(cachePath);
       assert.notDeepEqual(filesBefore, []);
-      await cache.clean(undefined, 0);
+      await cache.clean(undefined, -1);
       const filesAfter = await getAllFilesMatching(cachePath);
       assert.deepEqual(filesAfter, []);
     });
 
     it("should remove something with the max age set to some value", async () => {
+      await cache.set(randomUUID(), testValue);
       const filesBefore = await getAllFilesMatching(cachePath);
       assert.notDeepEqual(filesBefore, []);
-      // NOTE: We're waiting a little so that the file's atime is different
-      await new Promise((resolve) => setTimeout(resolve, 10));
-      await cache.set(randomUUID(), testValue);
-      await cache.clean(10);
+      await cache.clean(
+        new Date().getTime() -
+          (await getAccessTime(filesBefore[0])).getTime() -
+          1,
+      );
       const filesAfter = await getAllFilesMatching(cachePath);
-      assert.notDeepEqual(filesAfter, []);
+      // NOTE: It would be nice to check whether we removed some but not all files,
+      // but that check has proven to be flaky, especially on Intel macOS, due to
+      // its heavy reliance on timing of the operations.
       assert.notDeepEqual(filesAfter, filesBefore);
     });
 
     it("should remove something with the max size set to some value", async () => {
+      await cache.set(randomUUID(), testValue);
```

Comment:
This wasn't flaky before either. I just noticed the logic in this test can be simplified without compromising what we actually want to test here.